### PR TITLE
Chitter Challenge Attempt

### PR DESCRIPTION
**Pull request by Hannah-Frost**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,133 +1,9 @@
...
-SimpleCov.formatter = SimpleCov::Formatter::MultiFormatter.new([
-  SimpleCov::Formatter::Console,
-  # Want a nice code coverage website? Uncomment this next line!
-  # SimpleCov::Formatter::HTMLFormatter
-])
-SimpleCov.start
-```
-
-You can see your test coverage when you run your tests. If you want this in a graphical form, uncomment the `HTMLFormatter` line and see what happens!
+## What criteria does this program satisfy?
```
Comment (Xa1n) [inline README.md]: I like this explanation. Helps to understand what your program can do

File: app.rb [hunk 2]
```diff
@@ -2,4 +2,18 @@
 
 class Chitter < Sinatra::Base
   enable :sessions
+
```
Comment (Xa1n) [inline app.rb]: Hard to tell by these commits whether you tested first and then wrote the code or vice versa (I'm not a TDD advocate though so I don't mind :))

File: README.md [hunk 3]
```diff
@@ -7,3 +7,19 @@
...
+3. Create the database(s) required by following the steps under the 'How to create the databases' heading.
+
+## How to create the databases
+1. `psql` : open your user database in the command line.
+2. `CREATE DATABASE chitter;` : create a database.
+3. `CREATE DATABASE chitter_test;` : create a database for testing.
+4. \c chitter; : connect to the chitter database.
+5. Access 01_create_chitter_table.sql located within db/migrations/. Run the query within both databases.
+
+## How to run the app
```
Comment (Xa1n) [inline README.md]: really good, clear instructions

File: lib/peep.rb [hunk 4]
```diff
@@ -1,6 +1,11 @@
 require 'pg'
 
 class Peep
+  attr_reader :post
```
Comment (Xa1n) [inline lib/peep.rb]: can you explain why you need an attr_reader here?

File: spec/units/peep_spec.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    end
+  end
+  describe '#add' do
+    it 'adds a post to the database' do
+      peep = Peep.new(post: 'Hello, Chitter!')
+
+      expect(peep.post).to eq 'Hello, Chitter!'
+    end
+  end
+end
```
Comment (Xa1n) [inline spec/units/peep_spec.rb]: I can understand everything you're trying to do in this commit, nice one :)

File: spec/features/peep_time_spec.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+feature 'peep has a time of creation' do
+  scenario 'peep is shown with time' do
+    visit '/home'
+    click_button 'New Peep'
+    fill_in 'post', with: "Hello, chitter!"
+    click_button 'Post'
+    time = Time.new.strftime("%d/%m %H:%M")
```
Comment (Xa1n) [inline spec/features/peep_time_spec.rb]: good use of the strftime method

File: lib/user.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    @name = name
+    @password = password
+    @email = email
+  end
+
+  def self.add(name:, password:, email:)
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
```
Comment (Xa1n) [inline lib/user.rb]: the class methods are getting really packed. Do you know a way to extract some of the code to another class?

File: spec/reset_test_database.rb [hunk 8]
```diff
@@ -3,4 +3,5 @@
 def reset_test_database
```
Comment (Xa1n) [inline spec/reset_test_database.rb]: good name for method

File: views/home.erb [hunk 9]
```diff
@@ -1,4 +1,8 @@
-<center><h4>________________________</h4>
+<% if @name.nil? %>
+<center><h1>You must sign in first.</h1></center>
+<% else %>
+<center><h4>Signed in as <%= @name %>.</h4>
```
Comment (Xa1n) [inline views/home.erb]: the css really helps with user interaction, good work :)

File: app.rb (hunk 2 above)
Comment (Hannah-Frost) [reply app.rb]: I usually only push to Github once my tests are green, so I understand that can be a little confusing. I'll try to do commit more frequently to better show my approach. 😎

File: lib/peep.rb (hunk 4 above)
Comment (Hannah-Frost) [reply lib/peep.rb]: That was to allow a test to have access to that particular method, it has since been updated though!

**Pull request closed without merging**